### ISSUE TITLE
"RuntimeError: Validation failure (0)" trying to run examples on Debian Stretch

### ISSUE THREAD

**Issue by cybervegan** [kitao/pyxel#23]
Running on Debian 9 (Stretch) with Python 3.6.

andy@broken-screen:~/pyxel_examples$ python3.6 02_jump_game.py 
Traceback (most recent call last):
  File "02_jump_game.py", line 168, in <module>
    App()
  File "02_jump_game.py", line 8, in __init__
    pyxel.init(160, 120, caption='Pyxel Jump Game')
  File "/home/andy/miniconda3/lib/python3.6/site-packages/pyxel/__init__.py", line 27, in init
    border_width, border_color)
  File "/home/andy/miniconda3/lib/python3.6/site-packages/pyxel/app.py", line 79, in __init__
    self._renderer = Renderer(width, height)
  File "/home/andy/miniconda3/lib/python3.6/site-packages/pyxel/renderer.py", line 28, in __init__
    DRAWING_FRAGMENT_SHADER)
  File "/home/andy/miniconda3/lib/python3.6/site-packages/pyxel/glwrapper.py", line 12, in __init__
    shaders.compileShader(fragment_shader, gl.GL_FRAGMENT_SHADER))
  File "/home/andy/miniconda3/lib/python3.6/site-packages/OpenGL/GL/shaders.py", line 196, in compileProgram
    program.check_validate()
  File "/home/andy/miniconda3/lib/python3.6/site-packages/OpenGL/GL/shaders.py", line 108, in check_validate
    glGetProgramInfoLog( self ),
RuntimeError: Validation failure (0): 

andy@broken-screen:~/pyxel_examples$ cat /etc/os-release 
PRETTY_NAME="Debian GNU/Linux 9 (stretch)"
NAME="Debian GNU/Linux"
VERSION_ID="9"
VERSION="9 (stretch)"
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"

**Comment by AndrewNelis** [comment]
Hi, I'm seeing the same traceback on Fedora 27 on the master branch of Pyxel. Python version 3.3.6.

I've attached [glxinfo.txt](https://github.com/kitao/pyxel/files/2257239/glxinfo.txt) in case that helps.

As above, the traceback message is empty so I can't see what's wrong.

**Comment by AndrewNelis** [comment]
Digging deeper, I added the `validate` argument to `compileProgram` and got a more useful error:

```python
    def __init__(self, vertex_shader, fragment_shader):
        try:
            self._program = shaders.compileProgram(
                shaders.compileShader(vertex_shader, gl.GL_VERTEX_SHADER),
                shaders.compileShader(fragment_shader, gl.GL_FRAGMENT_SHADER), validate=False)
        except Exception as e:
            print(str(e).encode('utf-8').decode('unicode_escape'))
            raise SystemExit()
        self._att = None
        self._tex_list = []
```
Error output is now:

```
Link failure (0): b'Vertex info
-----------
0(32) : warning C7050: "v_max_pos" might be used before being initialized
0(31) : warning C7050: "v_min_pos" might be used before being initialized

Fragment info
-------------
0(154) : error C5208: Sampler needs to be a uniform (global or parameter to main), need to inline function or resolve conditional expression
0(155) : error C5208: Sampler needs to be a uniform (global or parameter to main), need to inline function or resolve conditional expression
```

I believe it's complaining about an issue with the `DRAWING_FRAGMENT_SHADER`, the `blt()` and `text()` functions.

**Comment by AndrewNelis** [comment]
Based on the link failure, I narrowed it down to these lines:

```glsl
int col = int(texture2D(u_texture[img], uv).r * 255.0);
```
According to [this stackoverflow answer](https://stackoverflow.com/a/12031821/4623093), we shouldn't be able to do this:

> In GLSL 1.30 to 3.30, you can have sampler arrays, but with severe restrictions on the index. The index must be an integral constant expression. Thus, while you can declare a sampler array, you can't loop over it.

So I don't understand how this code was working ever? I suspect there's a disconnect in GL versions maybe? I worked around it by unwrapping the array access so the index is a constant and now things work:

https://gist.github.com/AndrewNelis/ea5d8780c2d6aab047b00f3f2d52aa7b

(can't attach patches to github commends :-1: )

Improvements welcome...

**Comment by kitao** [comment]
I see.
I shouldn't use variables to specify the index of the texture image array in GLSL.

I'll change to use if as your patch.

**Comment by kitao** [comment]
Thank you!
I fixed this bug in Pyxel 0.7.5.

**Comment by AndrewNelis** [comment]
Thanks for patching.
